ac-local-plugin/code/LocalLlama.cpp

```diff
@@ -125,138 +155,196 @@ class ChatSession {
     }
 };
 
+SessionCoro<void> Llama_runInstance(coro::Io io, std::unique_ptr<llama::Instance> instance) {
+    using Schema = sc::StateInstance;
 
-class LlamaInstance final : public Instance {
-    std::shared_ptr<llama::Model> m_model;
-    llama::Instance m_instance;
+    struct Runner : public BasicRunner {
+        llama::Instance& m_instance;
+        std::optional<ChatSession> m_chatSession;
 
-    std::optional<ChatSession> m_chatSession;
+        Runner(llama::Instance& instance)
+            : m_instance(instance)
+        {
+            schema::registerHandlers<Schema::Ops>(m_dispatcherData, *this);
+        }
 
-    schema::OpDispatcherData m_dispatcherData;
-public:
-    using Schema = ac::schema::LlamaCppProvider::InstanceGeneral;
-    using Interface = ac::schema::LlamaCppInterface;
+        Schema::OpRun::Return on(Schema::OpRun, Schema::OpRun::Params&& params) {
+            if (m_chatSession) {
+                throw_ex{} << "llama: chat already started";
+            }
```

Comment:
chat can be solved as a deeper coroutine, as opposed to having a chat or not

Reply:
fed